ultralytics/tracker/track.py

```diff
@@ -39,8 +39,7 @@ def on_predict_start(predictor, persist=False):
 def on_predict_postprocess_end(predictor):
     """Postprocess detected boxes and update with object tracking."""
     bs = predictor.dataset.bs
-    im0s = predictor.batch[2]
-    im0s = im0s if isinstance(im0s, list) else [im0s]
+    im0s = predictor.batch[1]
```

Comment:
I was debugging this problem and found this solution.😄

```python
    def applySparseOptFlow(self, raw_frame, detections=None):
        # ...
        
        print(raw_frame) # < cv2.VideoCapture 000002B9F9099690>
        
        height, width, _ = raw_frame.shape # AttributeError: 'cv2.VideoCapture' object has no attribute 'shape'
        frame = cv2.cvtColor(raw_frame, cv2.COLOR_BGR2GRAY)
        H = np.eye(2, 3)
        
        # ...
```

https://github.com/ultralytics/ultralytics/blob/44c7c3514d87a5e05cfb14dba5a3eeb6eb860e70/ultralytics/tracker/utils/gmc.py#L245-L250

Reply:
@wangzexi that's because we updated stream_loader recently, which made `im0s` from `batch[2]` to `batch[1]` so there would be img issues if we stick with `batch[2]`.